main.go

```diff
@@ -43,6 +53,35 @@ func populateRequestBody(rw *prometheus.WriteRequest, r *http.Request) error {
 	return nil
 }
 
+func getAuthenticationToken(r *http.Request) string {
+	if val, ok := r.Header[headerAuthorization]; ok {
+		return val[0]
+	}
+	return ""
+}
+
+func validateToken(url *url.URL, clusterId string, token string) error {
```

Comment:
this could be enhanced by hooking in caching mechanism to reduce traffic sent to KFM.